src/sentry/utils/redis.py

```diff
@@ -103,13 +103,12 @@ def factory(self, **config):
         hosts = config.get('hosts')
         hosts = hosts.values() if isinstance(hosts, dict) else hosts
 
-        # Redis cluster does not wait to attempt to connect, we don't want the
-        # application to fail to boot because of this, raise a KeyError
-        try:
+        # Redis cluster does not wait to attempt to connect. We'd prefer to not
+        # make TCP connections on boot. Wrap the client in a lazy proxy object.
+        def cluster_factory():
             return RetryingStrictRedisCluster(startup_nodes=hosts, decode_responses=True)
-        except RedisClusterException:
-            logger.warning('Failed to connect to Redis Cluster', exc_info=True)
-            raise KeyError('Redis Cluster could not be initalized')
+
+        return SimpleLazyObject(cluster_factory)
```

Comment:
You could have just used a lambda. :)

```python
return SimpleLazyObject(lambda: RetryingStrictRedisCluster(startup_nodes=hosts, decode_responses=True))
```

Reply:
I considered that, but I liked the named function a little bit more. I can switch it over if you think it falls more in line without our satndards.

Reply:
Nah, I wasn't really suggesting to change anything. Just my preference to use a lambda for this, but that's my subjective opinion.